.github/workflows/ci-fork.yml

```
@@ -0,0 +1,62 @@
+name: CI (Fork)
+
+on:
+  workflow_run:
+    workflows: ["CI"]
+    types:
+      - completed
+
+jobs:
+  unit-test-results:
+    name: Unit Test Results
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.conclusion != 'skipped' &&
+      github.event.workflow_run.head_repository.full_name != github.repository
+
+    steps:
+      - name: Download Artifacts
+        uses: actions/github-script@v3
+        with:
+          script: |
+            var fs = require('fs');
+            var path = require('path');
+            var artifacts_path = path.join('${{github.workspace}}', 'artifacts')
+            fs.mkdirSync(artifacts_path, { recursive: true })
+
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{ github.event.workflow_run.id }},
+            });
+
+            for (const artifact of artifacts.data.artifacts) {
+               var download = await github.actions.downloadArtifact({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  artifact_id: artifact.id,
+                  archive_format: 'zip',
+               });
+               var artifact_path = path.join(artifacts_path, `${artifact.name}.zip`)
+               fs.writeFileSync(artifact_path, Buffer.from(download.data));
+               console.log(`Downloaded ${artifact_path}`);
+            }
+      - name: Extract Artifacts
+        run: |
+          for file in artifacts/*.zip
+          do
+            if [ -f "$file" ]
+            then
+              dir="${file/%.zip/}"
+              mkdir -p "$dir"
+              unzip -d "$dir" "$file"
+            fi
+          done
+
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        with:
+          check_name: unit-test-results
+          comment_title: Unit Test Results
+          files: "artifacts/*/**/*.xml"
+          commit: ${{ github.event.workflow_run.head_sha }}
```

The artifacts URL is available in the event, so we could simplify this whole job like this:

```suggestion
    steps:
      - name: Download Artifacts
        run: |
          gh api $(gh api ${{ github.event.workflow_run.artifacts_url }} -q '.artifacts[] | select(.name=="unit-test-results") .archive_download_url') > unit-test-results.zip
          unzip unit-test-results.zip

      - name: Publish unit test results
        uses: EnricoMi/publish-unit-test-result-action@v1
        with:
          check_name: unit-test-results
          comment_title: Unit Test Results
          files: "**/target/surefire-reports/TEST-*.xml"
          commit: ${{ github.event.workflow_run.head_sha }}
```

---

Let me look into this, that would be great.

---

I have created a PR to your action with a similar change. Here is an example run: https://github.com/jgiannuzzi/publish-unit-test-result-action/runs/2664871818?check_suite_focus=true

---

I did not know `gh` CLI is available, much more elegant. Will add this to the `publish-unit-test-result` action docs :-D